components/core/src/clp_s/search/OutputHandler.cpp

```diff
@@ -4,45 +4,63 @@ namespace clp_s::search {
 ResultsCacheOutputHandler::ResultsCacheOutputHandler(
         std::string const& uri,
         std::string const& collection,
-        uint64_t batch_size
+        uint64_t batch_size,
+        uint64_t max_num_results,
+        bool output_timestamp
 )
-        : m_batch_size(batch_size) {
+        : OutputHandler(output_timestamp),
+          m_batch_size(batch_size),
+          m_max_num_results(max_num_results) {
     try {
         auto mongo_uri = mongocxx::uri(uri);
         m_client = mongocxx::client(mongo_uri);
         m_collection = m_client[mongo_uri.database()][collection];
+        m_results.reserve(m_batch_size);
     } catch (mongocxx::exception const& e) {
         throw OperationFailed(ErrorCode::ErrorCodeBadParamDbUri, __FILENAME__, __LINE__);
     }
 }
 
 void ResultsCacheOutputHandler::flush() {
+    size_t count = 0;
+    while (false == m_latest_results.empty()) {
```

Comment:
Does this mean that we output up to m_max_num_results after scanning each table?

Could we potentially output a smaller subset of the most recent results here (just so the user can see something quickly) instead of dumping  all of them? My concern is that this can be wasted work if we later find some more recent results in the same archive.

Reply:
Right, we can send latest N results of each table and let mongodb sort and find latest results (just what we did for walmart). In theory,  we could send up to N*M (M is the number of tables) results. But as the size of table is highly skewed, we send far less results. The only drawback is that the UI may display some incorrect results at the beginning, and then refresh them.

Reply:
Right, we can send latest N results of each table and let mongodb sort and find latest results. In theory, we could send up to N*M (M is the number of tables) results. But as the size of table is highly skewed, we send far less results. The only drawback is that the UI may display some incorrect results at the beginning, and then refresh them.